### ISSUE TITLE
Composer Auto-Loader Not PHP5.2 Compatible

### ISSUE THREAD

**Issue by huyingjie** [benbalter/wordpress-github-sync#79]
I downloaded the plugin from WordPress.org

plugin version: 1.1.1
WordPress 4.3

I tried to install the plugin in two websites. Both of them cannot activate the plugin

When I activate the plugin, it gives error

~~~
Plugin could not be activated because it triggered a fatal error.

Warning: require_once(__DIR__/composer/autoload_real.php) [function.require-once]: failed to open stream: No such file or directory in /websiteName/wordpress/wp-content/plugins/wp-github-sync/vendor/autoload.php on line 5

Fatal error: require_once() [function.require]: Failed opening required '__DIR__/composer/autoload_real.php' (include_path='.:/opt/php52/lib/php') in /websitename.com/wordpress/wp-content/plugins/wp-github-sync/vendor/autoload.php on line 5
~~~

It was successfully activated and functions well in the website built in my laptop.  

I really like your plugin after trying on my local computer, and hope that this problem can be solved.

**Comment by huyingjie** [comment]
I found the answer. My server used PHP 5.2. I upgraded it to PHP5.4 then the problem solved. According to this page <https://github.com/picocms/Pico/issues/65>, PHP 5.3 is sufficient to solve it.

**Comment by mAAdhaTTah** [comment]
Actually, this is kind of an interesting point: Composer's autoloader isn't compatible with PHP5.2. We removed the closure in order to make this plugin 5.2 compatible, but if Composer blocks it, it's all for naught.

There is a package to create a 5.2-compatible autoloader: https://bitbucket.org/xrstf/composer-php52 - may be worth using.

**Comment by mAAdhaTTah** [comment]
Also, the `__DIR__` constant isn't available in PHP 5.2.
